### PR TITLE
(For David) $app and $browsers to more events

### PR DESCRIPTION
**Pull request by hd**

### CODE REVIEW

**Comment by ehrmann** [comment]
The tests are repeating themselves a lot. Maybe you can accomplish the same thing with a parameterized test, but I also wouldn't mind a robust `BaseAppBrowserFieldSetTest` and spot-checks on each event class, or just making sure the event class extends `BaseAppBrowserFieldSetTest`.

File: src/test/java/com/siftscience/BaseAppBrowserFieldSetTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        // Verify the response.
+        Assert.assertEquals(HTTP_OK, siftResponse.getHttpStatusCode());
+        Assert.assertEquals(0, (int) siftResponse.getBody().getStatus());
+        JSONAssert.assertEquals(response.getBody().readUtf8(),
+            siftResponse.getBody().toJson(), true);
+
+        server.shutdown();
+    }
+
+    private class TestFieldSet extends BaseAppBrowserFieldSet<TestFieldSet> {
```
Comment (ehrmann) [inline src/test/java/com/siftscience/BaseAppBrowserFieldSetTest.java]: nit: `static`

File: src/test/java/com/siftscience/AddItemToCartEventTest.java [hunk 2]
```diff
@@ -11,40 +13,41 @@
 
 import static java.net.HttpURLConnection.HTTP_OK;
 
-public class AddItemToCardEventTest {
```
Comment (ehrmann) [inline src/test/java/com/siftscience/AddItemToCartEventTest.java]: Was this class just renamed? I'm not sure if you use the added imports

Comment (hd) [reply src/test/java/com/siftscience/AddItemToCartEventTest.java]: Class was renamed. I removed unused imports.

File: src/test/java/com/siftscience/BaseAppBrowserFieldSetTest.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        "\"$type\"      : \"$test\",\n" +
+        "\"$api_key\"   : \"" + DUMMY_API_KEY + "\",\n" +
+        "\"$test_field\": \"" + DUMMY_TEST_FIELD + "\",\n" +
+        "\"$user_id\"   : \"" + DUMMY_USERID + "\"" +
+        "%s" +
+        "\n}";
+
+    @Test
+    public void testAllSubclasses() {
+        Class<?> [] subclasses = {
```
Comment (ehrmann) [inline src/test/java/com/siftscience/BaseAppBrowserFieldSetTest.java]: Where' are CreateOrderFieldSet, LogoutFieldSet, RemoveItemFromCartFieldSet, and UpdateOrderFieldSet?

Comment (hd) [reply src/test/java/com/siftscience/BaseAppBrowserFieldSetTest.java]: Oh good catch. `LogoutFieldSet` doesn't have `$app` nor `$browser`.

**Pull request merged**